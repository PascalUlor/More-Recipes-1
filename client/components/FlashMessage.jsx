import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import classnames from 'classnames';


/**
 * @description HOC for FlashMessage component
 *
 * @class FlashMessage
 *
 * @extends Component
 */
export class FlashMessage extends Component {
  /**
   * @description displays succes or error flash message
   *
   * @returns { jsx } jsx - renders FlashMessage components
   */
  render() {
    const { message } = this.props;
    return (
      <div className={classnames('text-center alert', {
        'alert-success': message.type === 'Success',
        'alert-danger': message.type === 'error'
      })}>
        {message.text}
      </div>
    );
  }
}

FlashMessage.propTypes = {
  message: PropTypes.shape({
    type: PropTypes.string,
    text: PropTypes.string
  }).isRequired
};
/**
 * @description maps redux state to props
 *
 * @param { object } state - holds flash message state
 *
 * @return { object } props - returns mapped props from state
 */
const mapStateToProps = state => ({
  message: state.flashMessage
});

export default connect(mapStateToProps)(FlashMessage);
