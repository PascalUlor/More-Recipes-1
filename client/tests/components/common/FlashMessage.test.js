import React from 'react';
/* eslint-disable */
import toJson from 'enzyme-to-json';
import { FlashMessage } from '../../../components/FlashMessage.jsx';


describe('<FlashMessage/>', () => {
  const props = {
    message: {}
  };
  it('renders flash messsage component without crashing', () => {
    const shallowWrapper = shallow(<FlashMessage {...props}/>);
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
