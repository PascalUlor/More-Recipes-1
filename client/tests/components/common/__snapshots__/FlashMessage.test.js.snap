// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlashMessage/> renders flash messsage component without crashing 1`] = `
<div
  className="text-center alert"
/>
`;
